lib/resque-bus.rb

```diff
@@ -16,4 +16,4 @@ module ResqueBus
   extend ::ResqueBus::Deprecated
 end
 
-QueueBus.adapter = QueueBus::Adapters::Resque.new
+QueueBus.adapter = QueueBus::Adapters::Resque.new unless QueueBus.has_adapter?
```

Comment:
I wonder if this should freak out (`raise`)
This approach might have the effect of silently being a no-op of the whole system and some quite confused engineers.

Reply:
That's what it currently does and it prevents me from requiring both sidekiq-bus and resque-bus in the same ruby runtime, which is making migrating more difficult. This has lead me to chasing my tail in our other repos while trying to figure out why the adapter was switching on me silently (even after I removed the resque-bus require because we require resque-bus/web to show the bus subscriptions.

All in all, I find that the raising here is making the gems more confusing and more difficult to use as it puts up roadblocks, does implicit autoloading via resque-bus/web, or requires monkey patching (and still requires tracing through the code so that load order is correct).

Which this change, requiring a second bus adapter will not break the system.

Reply:
hmmm, if a log then.
```
if has_adapter?
  Logger.warning("ResqubeBus:: not using Resqbue because #{QueueBusAdapter.class.name} already set up."
else
  QueueBus.adapter = QueueBus::Adapters::Resque.new
end
```

Reply:
Good idea, I'll try that out.

Reply:
One issue that arises is that this is done during loading, so there isn't really a logger yet for QueueBus.

I'm gonna put in a ruby warning.